tools/ARIAtools/extractProduct.py

```diff
@@ -1297,13 +1403,19 @@ def finalize_metadata(outname, bbox_bounds, arrres, dem_bounds, prods_TOTbbox, d
     # get minimum version
     version_check = []
     for i in prod_list:
-        v_num = i.split(':')[-2].split('/')[-1]
-        v_num = v_num.split('.nc')[0][-5:]
+        if not nisar_file:
+            v_num = i.split(':')[-2].split('/')[-1]
+            v_num = v_num.split('.nc')[0][-5:]
+        else:
+            basename = os.path.basename(i.split('"')[1])
+            v_num = basename.split('_')[-1][:-3]
+            v_num = '.'.join(v_num)
         version_check.append(v_num)
     version_check = min(version_check)
 
     metadatalyr_name = outname.split('/')[-2]
-    if metadatalyr_name in GEOM_LYRS and version_check < '2_0_4':
+    if (metadatalyr_name in GEOM_LYRS and version_check < '2_0_4') \
+            and not nisar_file:
```

Comment:
prefer parens over backslash (pep8), the continued line should align with metadatalyr_name